src/API/OnboardingTasks.php

```diff
@@ -826,11 +839,39 @@ public function snooze_task( $request ) {
 			wc_admin_record_tracks_event( 'tasklist_remindmelater_task', array( 'task_name' => $task_id ) );
 		}
 
-		$snooze_task['isSnoozed']    = true;
-		$snooze_task['snoozedUntil'] = $snoozed_until;
-
-		return rest_ensure_response( $snooze_task );
+		return rest_ensure_response( $update );
 	}
 
+	/**
+	 * Undo snooze of a single task.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Request|WP_Error
+	 */
+	public function undo_snooze_task( $request ) {
+		$id = $request->get_param( 'id' );
+
+		$snoozed = get_option( 'woocommerce_task_list_remind_me_later_tasks', array() );
+
+		if ( ! isset( $snoozed[ $id ] ) ) {
+			return new \WP_Error(
+				'woocommerce_tasks_invalid_task',
+				__( 'Sorry, no snoozed task with that ID was found.', 'woocommerce-admin' ),
+				array(
+					'status' => 404,
+				)
+			);
+		}
+
+		unset( $snoozed[ $id ] );
+
+		$update = update_option( 'woocommerce_task_list_remind_me_later_tasks', $snoozed );
+
+		if ( $update ) {
+			wc_admin_record_tracks_event( 'tasklist_undo_remindmelater_task', array( 'task_name' => $id ) );
+		}
+
+		return rest_ensure_response( $update );
```

Comment:
I know I did the same thing in my dismiss task, but I like the returned task as a response that you did previously.  

Maybe we should update all of these to return the updated task.  Thoughts?  Can be done here or a follow-up if we want to implement.